src/content/docs/fr/2/guide/upgrade-migrate/from-tauri-1.mdx

```diff
@@ -0,0 +1,1148 @@
+            .build()?;
+        Ok(())
+    })
+```
+
+#### Utilisez `tauri::menu::PredefinedMenuItem`
+
+Utilisez `tauri::menu::PredefinedMenuItem` à la place de `tauri::MenuItem`:
+
+```rust
+use tauri::menu::{MenuBuilder, PredefinedMenuItem};
+
+tauri::Builder::default()
+    .setup(|app| {
+        let menu = MenuBuilder::new(app).item(&PredefinedMenuItem::copy(app)).build()?;
+        Ok(())
+    })
+```
+
+:::tip
+Le constructeur de menu possède des méthodes dédiées pour ajouter chaque "item" de menu prédéfini pour que vous puissiez appeler `.copy()` à la place de `.item(&PredefinedMenuItem::copy(app, None))`.
+:::
+
+#### Utilisez `tauri::menu::MenuItemBuilder`
+
+Utilisez `tauri::menu::MenuItemBuilder` à la place de `tauri::CustomMenuItem`:
+
+```rust
+use tauri::menu::MenuItemBuilder;
+
+tauri::Builder::default()
+    .setup(|app| {
+        let toggle = MenuItemBuilder::new("Toggle").accelerator("Ctrl+Shift+T").build(app);
+        Ok(())
+    })
+```
+
+#### Utilisez `tauri::menu::SubmenuBuilder`
+
+Utilisez `tauri::menu::SubmenuBuilder` à la place de `tauri::Submenu`:
+
+```rust
+use tauri::menu::{MenuBuilder, SubmenuBuilder};
+
+tauri::Builder::default()
+    .setup(|app| {
+        let submenu = SubmenuBuilder::new(app, "Sub")
+            .text("Tauri")
+            .separator()
+            .check("Is Awesome")
+            .build()?;
+        let menu = MenuBuilder::new(app).item(&submenu).build()?;
+        Ok(())
+    })
+```
+
+`tauri::Builder::menu` prend maintenant une "closure" car le menu a besoin d'une instance de "Manager" pour être construit. Voir [la documentation](https://docs.rs/tauri/2.0.0-alpha/tauri/struct.Builder.html#method.menu) pour plus d'information.
+
+#### Evenement du Menu
+
+L'API Rust `tauri::Builder::on_menu_event` a été retirée. Utilisez `tauri::App::on_menu_event` ou `tauri::AppHandle::on_menu_event` à la place:
+
+```rust
+use tauri::menu::{CheckMenuItemBuilder, MenuBuilder, MenuItemBuilder};
+
+tauri::Builder::default()
+    .setup(|app| {
+        let toggle = MenuItemBuilder::with_id("toggle", "Toggle").build(app);
+        let check = CheckMenuItemBuilder::new("Mark").build(app);
+        let menu = MenuBuilder::new(app).items(&[&toggle, &check]).build()?;
+
+        app.set_menu(menu)?;
+
+        app.on_menu_event(move |app, event| {
+            if event.id() == check.id() {
+                println!("`check` triggered, do something! is checked? {}", check.is_checked().unwrap());
+            } else if event.id() == "toggle" {
+                println!("toggle triggered!");
+            }
+        });
+        Ok(())
+    })
+```
+Notez qu'il y a deux manières de vérifier que "item" de menu est séléctioné: on peut "move" (la fonctionnalités rust des closure) l'"item" vers un gestionnaire d'événement (event handler) sous forme de closure et comparer les IDs, ou d"finir une ID personnalisée via le constructeur `with_id` et utiliser cette ID (chaine de caractère) pour la comparer

```

Comment:
```suggestion
Notez qu'il y a deux manières de vérifier que "item" de menu est sélectionné: on peut "move" (la fonctionnalités rust des closure) l'item" vers un gestionnaire d'événement (event handler) sous forme de closure et comparer les IDs, ou définir une ID personnalisée via le constructeur `with_id` et utiliser cette ID (chaine de caractère) pour la comparer.
```

Reply:
On `l'"item"` I'm pretty sure that a single quote can't replace a double, fixed others mistakes